+nosnoc/+dcs/Gcs.m

```diff
@@ -0,0 +1,67 @@
+classdef Gcs < nosnoc.dcs.Base
+    properties
+        lambda
+
+        f_x
+
+        dims
+
+        f_fun
+        nabla_c_fun
+        c_fun
+        g_comp_path_fun
+    end
+
+    methods
+        function obj = Gcs(model)
+            obj.model = model;
+            obj.dims = model.dims;
+        end
+        
+        function generate_variables(obj, opts)
+            import casadi.*
+            dims = obj.dims;
+            % dimensions
+            dims.n_lambda = dims.n_c;
+            obj.lambda = define_casadi_symbolic(opts.casadi_symbolic_mode,'lambda',dims.n_lambda);
+
+            obj.dims = dims;
+        end
+
+        function generate_equations(obj, opts)
+            import casadi.*
+            model = obj.model;
+            dims = obj.dims;
+
+            nabla_c = model.c.jacobian(model.x)';
+            
+            obj.f_x = model.f_x + model.E*nabla_c*obj.lambda;
+
+            obj.f_x_fun = Function('f_x', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {obj.f_x, model.f_q});
```

Comment:
if you have f_q_fun below, is it needed that you have f_q as output. 
I guess the first option is when you want to have a accurate integration of the objective (default), the second is for special cases like state estimation, right? (a one line comment in the code would be helpful for our future self)

Reply:
Yea this is not actually used in the discrete time problem, and was left purely due to copy paste. Removing now.